vncviewer/UserDialog.cxx

```diff
@@ -187,15 +217,19 @@ void UserDialog::getUserPasswd(bool secure, char** user, char** password)
   while (win->shown()) Fl::wait();
 
   if (ret_val == 0) {
-    if (user)
+    if (user) {
       *user = strDup(username->value());
+      user_ = strDup(username->value());
```

Comment:
You have to take care here with the difference between `strdup()` and `strDup()`. They use different allocators.

Reply:
Thanks, did not realize this. Changed all occurrences to `strDup()` and `delete []`